### PR TITLE
style(ui): add dark mode social icon

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add dark mode social icon.

image

image

## Testing

test locally
@logto-io/eng

### CODE REVIEW

File: packages/ui/src/components/Button/SocialIconButton.tsx [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-  connector: ConnectorData;
+  logo: string;
+  target?: string;
   onClick?: () => void;
 };
 
-const SocialIconButton = ({ className, connector, onClick }: Props) => {
-  const { target, logo } = connector;
+const SocialIconButton = ({ className, logo, target, onClick }: Props) => {
+  const inverseBackground = target === 'apple';
```
Comment (gao-sun) [inline packages/ui/src/components/Button/SocialIconButton.tsx:14]: apple is our daddy

File: packages/ui/src/containers/SocialSignIn/SocialSignInList/index.tsx [hunk 2]
```diff
@@ -39,17 +39,23 @@ const SocialSignInList = ({
...
-      ))}
+      {displayConnectors.map((connector) => {
+        const { id, name, logo, logoDark, target } = connector;
+
+        return (
+          <SocialLinkButton
+            key={id}
+            className={styles.socialLinkButton}
+            name={name}
+            logo={theme === 'dark' ? logoDark ?? logo : logo}
```
Comment (gao-sun) [inline packages/ui/src/containers/SocialSignIn/SocialSignInList/index.tsx:50]:
```suggestion
            logo={theme === 'dark' && logoDark || logo}
```

File: packages/ui/src/containers/SocialLanding/index.tsx [hunk 3]
```diff
@@ -13,13 +13,17 @@ type Props = {
...
   const connector = experienceSettings?.socialConnectors.find(({ id }) => id === connectorId);
 
   return (
     <div className={classNames(styles.container, className)}>
       <div className={styles.connector}>
-        {connector?.logo ? <img src={connector.logo} /> : connectorId}
+        {connector ? (
+          <img src={theme === 'dark' ? connector.logoDark ?? connector.logo : connector.logo} />
+        ) : (
+          connectorId
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/ui/src/containers/SocialLanding/index.tsx:25]: As the connector id comes from the URL params, not from a trusted source, the connector instance might not exist. i.e. I randomly add some connector Id to the URL.

File: packages/ui/src/containers/SocialSignIn/SocialSignInIconList/index.tsx [hunk 4]
```diff
@@ -24,20 +24,25 @@ const SocialSignInIconList = ({
...
-        />
-      ))}
+      {connectors.map((connector) => {
+        const { id, target, logo, logoDark } = connector;
+
+        return (
+          <SocialIconButton
+            key={id}
+            className={styles.socialButton}
+            logo={theme === 'dark' && target !== 'apple' ? logoDark ?? logo : logo}
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/ui/src/containers/SocialSignIn/SocialSignInIconList/index.tsx]: updated

File: packages/ui/src/containers/SocialSignIn/SocialSignInList/index.tsx (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/containers/SocialSignIn/SocialSignInList/index.tsx]: updated

File: packages/ui/src/components/Button/SocialIconButton.tsx [hunk 5]
```diff
@@ -1,24 +1,28 @@
...
-    <button className={classNames(styles.socialButton, className)} onClick={onClick}>
-      {logo && <img src={logo} alt={target} className={styles.icon} />}
-    </button>
-  );
-};
+const SocialIconButton = ({ className, logo, target, onClick }: Props) => (
+  <button
+    className={classNames(
+      styles.socialButton,
+      isAppleConnector('target') && styles.inverse,
```
Comment (wangsijie) [inline packages/ui/src/components/Button/SocialIconButton.tsx:19]: `isAppleConnector(target)`

**Review by wangsijie — APPROVED** [review]
LGTM

**Pull request merged**